app/javascript/components/upload/UploadWizard.jsx

```diff
@@ -357,67 +398,85 @@ export function RawUploadWizard({ studyAccession, name }) {
       response.files.forEach(file => formatFileFromServer(file))
+      setIsAnnDataExperience(response.files?.filter(AnnDataFileFilter)?.length > 0 && response.feature_flags?.ingest_anndata_file)
       setServerState(response)
       setFormState(_cloneDeep(response))
       setTimeout(pollServerState, POLLING_INTERVAL)
     })
+
     window.document.title = `Upload - Single Cell Portal`
   }, [studyAccession])
 
   const nextStep = STEPS[currentStepIndex + 1]
   const prevStep = STEPS[currentStepIndex - 1]
+
+  /**
+   * Returns the appropriate content to display for the UploadWizard
+   * @param  formState
+   * @returns The content for the upload wizard, either the steps for upload or the split view for choosing the data upload experience
+   */
+  function getWizardContent(formState, serverState) {
+    if (!formState?.files.length && !choiceMade && serverState?.feature_flags?.ingest_anndata_file) {
+      return <UploadExperienceSplitter {...{ setIsAnnDataExperience, setChoiceMade }}/>
+    } else if (choiceMade || formState?.files.length || !serverState?.feature_flags?.ingest_anndata_file) {
+      return <> <div className="row wizard-content">
+        <div>
+          <WizardNavPanel {...{
+            formState, serverState, currentStep, setCurrentStep, studyAccession, mainSteps: MAIN_STEPS,
+            supplementalSteps: SUPPLEMENTAL_STEPS, nonVizSteps: NON_VISUALIZABLE_STEPS, studyName: name, isAnnDataExperience
+          }} />
+        </div>
+        <div id="overflow-x-scroll">
+          <div className="flexbox-align-center top-margin left-margin">
+            <h4>{currentStep.header}</h4>
+            <div className="prev-next-buttons">
+              { prevStep && <button
+                className="btn terra-tertiary-btn margin-right"
+
+                onClick={() => setCurrentStep(prevStep)}>
+                <FontAwesomeIcon icon={faChevronLeft}/> Previous
+              </button> }
+              { nextStep && <button
+                className="btn terra-tertiary-btn"
+                onClick={() => setCurrentStep(nextStep)}>
+            Next <FontAwesomeIcon icon={faChevronRight}/>
+              </button> }
+            </div>
+          </div>
+          <div>
+            <currentStep.component
+              setCurrentStep={setCurrentStep}
+              formState={formState}
+              serverState={serverState}
+              deleteFile={deleteFile}
+              updateFile={updateFile}
+              saveFile={saveFile}
+              addNewFile={addNewFile}
+              isAnnDataExperience={isAnnDataExperience}
+              deleteFileFromForm={deleteFileFromForm}
+            />
+          </div>
+        </div>
+      </div></>
+    }
+  }
+
   return (
     <StudyContext.Provider value={studyObj}>
+      {/* If the formState hasn't loaded show a spinner */}
+      {(!formState && !serverState) && <LoadingSpinner testId="upload-wizard-spinner"/>
+      }
```

Comment:
Helpful comment, and a nice UI feature!

For later, maybe:

From the [demo video](https://docs.google.com/presentation/d/1QzHy48am_lqk8T3rFSL-4OmIRyaTTK_B9ozZC6HjEVs/edit#slide=id.g216c356411d_0_14) (starting around 0:42), I see the loading spinner is small and at the upper-leftmost corner of the page.  I think it'd look better 2-5x larger, and centered on the page.

<img width="1316" alt="Small_spinner_at_upper-leftmost_corner__SCP_2023-03-08" src="https://user-images.githubusercontent.com/1334561/223808286-2c321f96-d882-498f-b559-eb4254e12a45.png">